client-app/core/api/graphql/push-messages/mutations/markAllPushMessagesRead/index.ts

```diff
@@ -0,0 +1,34 @@
+import { useMutation } from "@/core/api/graphql/composables/useMutation";
+import { MarkAllPushMessagesReadDocument, OperationNames } from "@/core/api/graphql/types";
+import type { GetPushMessagesQuery } from "@/core/api/graphql/types";
+
+export function useMarkAllPushMessagesRead() {
+  return useMutation(MarkAllPushMessagesReadDocument, {
+    // TODO: Remove all code below in next iteration when XAPI will return objects from mutations
+    optimisticResponse: {
+      markAllPushMessagesRead: true,
+    },
+    updateQueries: {
+      [OperationNames.Query.GetPushMessages]: (previousQueryResult, { mutationResult }) => {
+        if (mutationResult.data?.markAllPushMessagesRead) {
+          const pushMessagesQueryResult = previousQueryResult as GetPushMessagesQuery;
+          return {
+            ...pushMessagesQueryResult,
+            // TODO: Move this code to optimisticResponse in next iteration for better UX responsitibility
+            pushMessages: {
+              unreadCount: 0,
+              items: pushMessagesQueryResult.pushMessages.items.map((pushMessage) => ({
+                ...pushMessage,
+                status: "Read",
+              })),
+            },
+          };
+        } else {
+          return { ...previousQueryResult };
+        }
+      },
+    },
+    // Just in case we did something wrong in cache
```

Comment:
Why should we? Why not always refetch them with proper cache policy?

Reply:
We already do refetch, but if we don't use manual cache update & `optimisticResponse` it looks like system "hang up" for a second or two. When XAPI start to return updated objects, we will need to leave just a few lines of `optimisticResponse` and remove everything else.